### ISSUE TITLE
saveJsonConfig metrics Differ From Booster Internal Metrics

### ISSUE THREAD

**Issue by acalejos** [dmlc/xgboost#9245]
When using the C API `XGBoosterSaveJsonConfig` function, the outputted JSON configuration metrics differ from those loaded in the actual model. The default values are not stored in the configuration even when the `disable_default_eval_metric` flag is not `true`, and only the last metric that was set with `XGBoosterSetParam` is shown, but is repeated the number of times that the parameter `eval_metric` was set. So for example, if I set `eval_metric` 4 times in the order "rmse","logloss","error","auc", then while I'm training and evaluate the model, it evaluates against all 4 metrics, but then when I dump the config I get "auc" listed 4 times.

Example training evaluation output (garbage data):
```
Iteration 0: %{"validation" => %{"error" => -0.07458905875682831, "logloss" => -16.04233488014766, "rmse" => 0.8611358094849999}}
Iteration 1: %{"validation" => %{"error" => -0.07458905875682831, "logloss" => -16.43325683048793, "rmse" => 0.7168819564974251}}
Iteration 2: %{"validation" => %{"error" => -0.07458905875682831, "logloss" => -16.622497345720017, "rmse" => 0.5988016411230994}}
```

Example Configuration Output:

```
%{
  "learner" => %{
    "generic_param" => %{
      "fail_on_invalid_gpu_id" => "0",
      "gpu_id" => "-1",
      "n_jobs" => "0",
      "nthread" => "0",
      "random_state" => "0",
      "seed" => "0",
      "seed_per_iteration" => "0",
      "validate_parameters" => "1"
    },
    "gradient_booster" => %{
      "gbtree_model_param" => %{"num_parallel_tree" => "1", "num_trees" => "3"},
      "gbtree_train_param" => %{
        "predictor" => "auto",
        "process_type" => "default",
        "tree_method" => "hist",
        "updater" => "grow_quantile_histmaker",
        "updater_seq" => "grow_quantile_histmaker"
      },
      "name" => "gbtree",
      "specified_updater" => false,
      "tree_train_param" => %{
        "alpha" => "0",
        "cache_opt" => "1",
        "colsample_bylevel" => "1",
        "colsample_bynode" => "1",
        "colsample_bytree" => "1",
        "eta" => "0.300000012",
        "gamma" => "0",
        "grow_policy" => "depthwise",
        "interaction_constraints" => "",
        "lambda" => "1",
        "learning_rate" => "0.300000012",
        "max_bin" => "256",
        "max_cat_threshold" => "64",
        "max_cat_to_onehot" => "4",
        "max_delta_step" => "0",
        "max_depth" => "6",
        "max_leaves" => "0",
        "min_child_weight" => "1",
        "min_split_loss" => "0",
        "monotone_constraints" => "()",
        "refresh_leaf" => "1",
        "reg_alpha" => "0",
        "reg_lambda" => "1",
        "sampling_method" => "uniform",
        "sketch_ratio" => "2",
        "sparse_threshold" => "0.20000000000000001",
        "subsample" => "1"
      },
      "updater" => %{"grow_quantile_histmaker" => %{}}
    },
    "learner_model_param" => %{
      "base_score" => "-7.458906E-2",
      "boost_from_average" => "1",
      "num_class" => "0",
      "num_feature" => "6",
      "num_target" => "1"
    },
    "learner_train_param" => %{
      "booster" => "gbtree",
      "disable_default_eval_metric" => "0",
      "multi_strategy" => "one_output_per_tree",
      "objective" => "reg:squarederror"
    },
    "metrics" => [
      %{"name" => "error"},
      %{"name" => "error"},
      %{"name" => "error"}
    ],
    "objective" => %{
      "name" => "reg:squarederror",
      "reg_loss_param" => %{"scale_pos_weight" => "1"}
    }
  },
  "version" => [2, 0, 0]
}
```

If I do one call to `XGBoosterEvalOneIter` then I can get the default metric value, but it does not solve the issue of the configuration only saving one of the metrics.

**Comment by trivialfis** [comment]
Thank you for raising the issue. The configuration for metrics is carried out twice due to historical reasons. I plan to resolve it along with https://github.com/dmlc/xgboost/issues/7693. Unfortunately, so far have yet to be able to prioritize it.

**Comment by acalejos** [comment]
@trivialfis FYI this appears to be fixed as of the 2.0.0 Release, at least in my testing.  Thank you for your hard work!
